### PR TITLE
Feat(storage): add aws s3 buckets per tenant

### PR DESCRIPTION
**Pull request by feiguoL**
**What this PR does**:
Add buckets per tenant that archive truly resource isolation

**Which issue(s) this PR fixes**:
Featrue #<https://github.com/grafana/loki/issues/2292>

**Checklist**
- [✔ ] Tests updated
- [✔ ] Documentation added
- [✔ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## master / unreleased
 
+* [FEATURE] Add buckets per tenant that archive truly resource isolation (`--s3.buckets-prefix` globally) each tenants will create one bucket with prefix. [#2292](https://github.com/grafana/loki/issues/2292)
```
Comment (pstibrany) [inline CHANGELOG.md:5]: Cortex release 1.8.0 is now in progress. Could you please rebase master and move the CHANGELOG entry under the master / unreleased section?

Comment (feiguoL) [reply CHANGELOG.md:5]: ok

Comment (feiguoL) [reply CHANGELOG.md:5]: > Cortex release 1.8.0 is now in progress. Could you please rebase master and move the CHANGELOG entry under the master / unreleased section?

done!

Comment (feiguoL) [reply CHANGELOG.md:5]: done!

Comment (jeromeinsf) [inline CHANGELOG.md:5]: [nit] achieve not archive

**Comment by LinTechSo** [comment]
Hi, any updates?

**Pull request closed without merging**